blogposts/2020/a-different-way-to-think-about-code-ownership.md

```diff
@@ -0,0 +1,79 @@
+
+## Exploring alternatives
+
+There was enough evidence that a problem existed to justify exploring alternatives.
+
+I found [Fullstory’s blog post about how they ran into similar problems with CODEOWNERS and ended up creating a bot](https://bionic.fullstory.com/taming-github-codeowners-with-bots/). I would have loved to experiment with using their bot, but unfortunately it was not open source.
+
+A few of our teammates, including myself, worked at Google so we were familiar with the [OWNERS](https://chromium.googlesource.com/chromium/src/+/master/docs/code_reviews.md) format used internally and in the Chromium project. This would have been an improvement over CODEOWNERS because multiple files are easier to maintain than a single large file and [Chromium’s implementation explicitly attempts to minimize the number of reviewers](https://chromium.googlesource.com/chromium/tools/depot_tools/+/master/owners.py#607), but I was worried this would continue to create a gatekeeper culture around modifying code.
+
+I wanted a solution that would:
+
+1. Allow developers to subscribe to the code changes that they cared about without creating an expectation that they would review all those changes.
+1. Allow authors to use their judgement and agency to intentionally select reviewers whose feedback they think would be valuable (for example: Who wrote the code being edited? Who reviewed the code being edited? Who provides valuable feedback in a timely manner?).
+1. Be easy to understand and maintain in a large and constantly evolving codebase.
+
+Neither CODEOWNERS nor OWNERs were designed with all these requirements in mind, so I started to think about what a new tool might look like, and I ended up with [Codenotify](https://github.com/sourcegraph/codenotify).
+
+## Designing Codenotify
+
+As the name implies, Codenotify is designed around the concept of notifications, not about establishing "ownership" of code or mandating "reviewers". Codenotify works on the command line and is codehost agnostic at its core, but it is also packaged as a GitHub Action so it is easy for repositories on GitHub, like ours, to adopt. The GitHub Action sends notifications to subscribers by mentioning them in a comment instead of adding them to the official "reviewers" or "assignees" lists on GitHub.
+
+Notification rules are stored in [CODENOTIFY files](https://sourcegraph.com/github.com/sourcegraph/codenotify/-/blob/README.md#codenotify-files) in any directory. This makes rules more maintainable and less brittle to moving code because the CODENOTIFY files are adjacent to the files they have rules for.
+
+Codenotify rules and files are familiar in syntax to CODEOWNERS, but simpler to understand because each rule is additive, not hierarchical. This means that you can understand how a single rule behaves without needing to consider any other rules (unlike OWNERS or CODEOWNERS).
+
+## Starting our experiment
```

Comment:
Maybe add a small section here with a CTA to readers to try it out. Can link to https://github.com/sourcegraph/codenotify.

Reply:
good idea, I wasn't sure about how best to format it and didn't see any examples of CTAs in recent posts so I just did a quote. lmk if you have a better idea.